sql-plugin/src/main/scala/com/nvidia/spark/rapids/HostAlloc.scala

```diff
@@ -186,6 +206,11 @@ private class HostAlloc(nonPinnedLimit: Long) extends HostMemoryAllocator with L
       allocAttemptFinishedWithoutException = true
     } finally {
       if (ret.isDefined) {
+        // Alternatively we could do the host watermark tracking in the JNI code to make
+        // it consistent with how we handle device memory tracking
```

Comment:
This seems like a note for a potential future improvement? Then can we make it clearer? I was confused for a bit whether this is about the current implementation or future improvement.

Reply:
It's not actually something I want to change,  I prefer how it is now, I just wanted to note that I had made this choice. I'll just remove the comment and it will be better served as a github comment here